[BUG]MultiplexerProxy panics when ResourceStore or getReqScope fails

**What happened**:

In `pkg/yurthub/proxy/multiplexer/multiplexerproxy.go`, `ServeHTTP` writes an error response when `ResourceStore()` or `getReqScope()` fails, but does not return afterward. Execution continues and type-asserts a nil `restStore`, which causes YurtHub to panic:

```go
restStore, err := sp.requestsMultiplexerManager.ResourceStore(gvr)
if err != nil {
    util.Err(errors.Wrapf(err, "failed to get rest storage"), w, r)
    // missing return
}

reqScope, err := sp.getReqScope(gvr)
if err != nil {
    util.Err(errors.Wrapf(err, "failed tp get req scope"), w, r)
    // missing return
}

lister := restStore.(rest.Lister)   // panic: nil interface
watcher := restStore.(rest.Watcher)
```

This is inconsistent with the correct pattern used a few lines above (lines 75–78), where `util.Err(...)` is followed by `return`.

**What you expected to happen**:

When `ResourceStore()` or `getReqScope()` returns an error, YurtHub should return a proper HTTP error response and stop handling the request. It should not panic.

**How to reproduce it (as minimally and precisely as possible)**:

1. Run this minimal repro against upstream master:

```go
package main

import (
	"net/http"
	"net/http/httptest"
	"os"

	apirequest "k8s.io/apiserver/pkg/endpoints/request"

	"github.com/openyurtio/openyurt/pkg/yurthub/kubernetes/meta"
	mproxy "github.com/openyurtio/openyurt/pkg/yurthub/proxy/multiplexer"
	"github.com/openyurtio/openyurt/pkg/yurthub/multiplexer"
)

func main() {
	tmpDir, _ := os.MkdirTemp("", "mp-repro")
	defer os.RemoveAll(tmpDir)

	restMapper, _ := meta.NewRESTMapperManager(tmpDir)
	mm := &multiplexer.MultiplexerManager{}
	sp := mproxy.NewMultiplexerProxy(mm, restMapper, make(<-chan struct{}))

	req := httptest.NewRequest(http.MethodGet, "/apis/nonexistent.group/v1/widgets", nil)
	info := &apirequest.RequestInfo{
		IsResourceRequest: true,
		APIGroup:          "nonexistent.group",
		APIVersion:        "v1",
		Resource:          "widgets",
		Verb:              "list",
	}
	req = req.WithContext(apirequest.WithRequestInfo(req.Context(), info))

	sp.ServeHTTP(httptest.NewRecorder(), req)
}
```

2. Observe panic:
```
panic: runtime error: invalid memory address or nil pointer dereference
```

Verified on upstream master at commit `81ca6625`.

**Anything else we need to know?**:

- **Impact:** Pool-scope metadata list/watch requests can crash YurtHub when storage or REST mapping is unavailable, instead of returning an HTTP error.
- **Suggested fix:** Add `return` after both `util.Err(...)` calls (lines 83 and 88).
- **Related code:** `util.Err` + `return` pattern already used correctly at lines 75–78 in the same file.

**Environment**:
- OpenYurt version: master (`81ca6625`)
- Kubernetes version: N/A (reproduced with unit-level handler test)
- OS: Linux
- Kernel: N/A
- Install tools: Go 1.24+
- Others: Reproduced with standalone Go program against local clone synced with upstream master

**others**

/kind bug

--- a/pkg/yurthub/proxy/multiplexer/multiplexerproxy.go
+++ b/pkg/yurthub/proxy/multiplexer/multiplexerproxy.go
@@ -86,7 +86,7 @@ func (sp *multiplexerProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	reqScope, err := sp.getReqScope(gvr)
 	if err != nil {
-		util.Err(errors.Wrapf(err, "failed tp get req scope"), w, r)
+		util.Err(errors.Wrapf(err, "failed to get req scope"), w, r)
 		return
 	}
 

--- a/pkg/yurthub/proxy/multiplexer/multiplexerproxy.go
+++ b/pkg/yurthub/proxy/multiplexer/multiplexerproxy.go
@@ -44,8 +44,13 @@ const (
 	minRequestTimeout = 300 * time.Second
 )
 
+type resourceMultiplexer interface {
+	Ready(gvr *schema.GroupVersionResource) bool
+	ResourceStore(gvr *schema.GroupVersionResource) (rest.Storage, error)
+}
+
 type multiplexerProxy struct {
-	requestsMultiplexerManager *multiplexer.MultiplexerManager
+	requestsMultiplexerManager resourceMultiplexer
 	restMapperManager          *hubmeta.RESTMapperManager
 	stop                       <-chan struct{}
 }

--- a/pkg/yurthub/proxy/multiplexer/multiplexerproxy.go
+++ b/pkg/yurthub/proxy/multiplexer/multiplexerproxy.go
@@ -81,11 +81,13 @@ func (sp *multiplexerProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	restStore, err := sp.requestsMultiplexerManager.ResourceStore(gvr)
 	if err != nil {
 		util.Err(errors.Wrapf(err, "failed to get rest storage"), w, r)
+		return
 	}
 
 	reqScope, err := sp.getReqScope(gvr)
 	if err != nil {
 		util.Err(errors.Wrapf(err, "failed tp get req scope"), w, r)
+		return
 	}
 
 	lister := restStore.(rest.Lister)

--- a/pkg/yurthub/proxy/multiplexer/multiplexerproxy_test.go
+++ b/pkg/yurthub/proxy/multiplexer/multiplexerproxy_test.go
@@ -20,6 +20,7 @@ package multiplexer
 import (
 	"bytes"
 	"context"
+	"errors"
 	"fmt"
 	"net/http"
 	"net/http/httptest"
@@ -36,6 +37,7 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
 	"k8s.io/apiserver/pkg/server"
 	"k8s.io/apiserver/pkg/storage"
 	"k8s.io/client-go/informers"
@@ -112,6 +114,23 @@ func (wr *wrapResponse) Write(buf []byte) (int, error) {
 	return l, err
 }
 
+type fakeResourceMultiplexer struct {
+	ready    bool
+	store    rest.Storage
+	storeErr error
+}
+
+func (f *fakeResourceMultiplexer) Ready(gvr *schema.GroupVersionResource) bool {
+	return f.ready
+}
+
+func (f *fakeResourceMultiplexer) ResourceStore(gvr *schema.GroupVersionResource) (rest.Storage, error) {
+	if f.storeErr != nil {
+		return nil, f.storeErr
+	}
+	return f.store, nil
+}
+
 func TestGetReqScopeReturnsErrorForUnknownGVR(t *testing.T) {
 	tmpDir, err := os.MkdirTemp("", "multiplexer-req-scope")
 	if err != nil {
@@ -130,6 +149,40 @@ func TestGetReqScopeReturnsErrorForUnknownGVR(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestMultiplexerProxyServeHTTPReturnsOnResourceStoreError(t *testing.T) {
+	tmpDir, err := os.MkdirTemp("", "multiplexer-resource-store")
+	if err != nil {
+		t.Fatalf("failed to make temp dir: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	restMapperManager, err := meta.NewRESTMapperManager(tmpDir)
+	if err != nil {
+		t.Fatalf("failed to create REST mapper manager: %v", err)
+	}
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	sp := &multiplexerProxy{
+		requestsMultiplexerManager: &fakeResourceMultiplexer{
+			ready:    true,
+			storeErr: errors.New("storage unavailable"),
+		},
+		restMapperManager: restMapperManager,
+		stop:              stopCh,
+	}
+
+	w := httptest.NewRecorder()
+	req := newEndpointSliceListRequest("/apis/discovery.k8s.io/v1/endpointslices", nil)
+
+	assert.NotPanics(t, func() {
+		sp.ServeHTTP(w, req)
+	})
+	assert.GreaterOrEqual(t, w.Code, http.StatusBadRequest)
+	assert.Contains(t, w.Body.String(), "failed to get rest storage")
+}
+
 func TestMultiplexerProxyServeHTTPReturnsOnGetReqScopeError(t *testing.T) {
 	tmpDir, err := os.MkdirTemp("", "multiplexer-serve-http")
 	if err != nil {
@@ -137,43 +190,28 @@ func TestMultiplexerProxyServeHTTPReturnsOnGetReqScopeError(t *testing.T) {
 	}
 	defer os.RemoveAll(tmpDir)
 
-	cfgRESTMapper, err := meta.NewRESTMapperManager(tmpDir)
+	restMapperManager, err := meta.NewRESTMapperManager(tmpDir)
 	if err != nil {
 		t.Fatalf("failed to create REST mapper manager: %v", err)
 	}
-	proxyRESTMapper, err := meta.NewRESTMapperManager(tmpDir)
-	if err != nil {
-		t.Fatalf("failed to create proxy REST mapper manager: %v", err)
-	}
-
-	clientset := fake.NewSimpleClientset()
-	factory := informers.NewSharedInformerFactory(clientset, 0)
-	poolScopeResources := []schema.GroupVersionResource{endpointSliceGVR}
-
-	healthChecker := fakeHealthChecker.NewFakeChecker(map[*url.URL]bool{})
-	loadBalancer := remote.NewLoadBalancer("round-robin", []*url.URL{}, nil, nil, healthChecker, nil, context.Background().Done())
-	dsm := multiplexerstorage.NewDummyStorageManager(mockCacheMap())
-	cfg := &config.YurtHubConfiguration{
-		PoolScopeResources:       poolScopeResources,
-		RESTMapperManager:        cfgRESTMapper,
-		SharedFactory:            factory,
-		LoadBalancerForLeaderHub: loadBalancer,
-	}
-	rmm := multiplexer.NewRequestMultiplexerManager(cfg, dsm, healthChecker)
 
 	stopCh := make(chan struct{})
 	defer close(stopCh)
-	if ok := cache.WaitForCacheSync(stopCh, func() bool { return rmm.Ready(&endpointSliceGVR) }); !ok {
-		t.Fatal("multiplexer manager is not ready")
+
+	sp := &multiplexerProxy{
+		requestsMultiplexerManager: &fakeResourceMultiplexer{ready: true},
+		restMapperManager:          restMapperManager,
+		stop:                       stopCh,
 	}
 
-	sp := NewMultiplexerProxy(rmm, proxyRESTMapper, stopCh)
 	w := httptest.NewRecorder()
-	req := newEndpointSliceListRequest("/apis/discovery.k8s.io/v1/endpointslices", nil)
+	req := newEndpointSliceListRequest("/apis/unknown.example.com/v1/widgets", nil)
 
 	assert.NotPanics(t, func() {
 		sp.ServeHTTP(w, req)
 	})
+	assert.GreaterOrEqual(t, w.Code, http.StatusBadRequest)
+	assert.Contains(t, w.Body.String(), "failed to get req scope")
 }
 
 func TestShareProxy_ServeHTTP_LIST(t *testing.T) {

--- a/pkg/yurthub/proxy/multiplexer/multiplexerproxy_test.go
+++ b/pkg/yurthub/proxy/multiplexer/multiplexerproxy_test.go
@@ -112,6 +112,69 @@ func (wr *wrapResponse) Write(buf []byte) (int, error) {
 	return l, err
 }
 
+func TestGetReqScopeReturnsErrorForUnknownGVR(t *testing.T) {
+	tmpDir, err := os.MkdirTemp("", "multiplexer-req-scope")
+	if err != nil {
+		t.Fatalf("failed to make temp dir: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	restMapperManager, err := meta.NewRESTMapperManager(tmpDir)
+	if err != nil {
+		t.Fatalf("failed to create REST mapper manager: %v", err)
+	}
+
+	sp := &multiplexerProxy{restMapperManager: restMapperManager}
+	unknownGVR := schema.GroupVersionResource{Group: "unknown.example.com", Version: "v1", Resource: "widgets"}
+	_, err = sp.getReqScope(&unknownGVR)
+	assert.Error(t, err)
+}
+
+func TestMultiplexerProxyServeHTTPReturnsOnGetReqScopeError(t *testing.T) {
+	tmpDir, err := os.MkdirTemp("", "multiplexer-serve-http")
+	if err != nil {
+		t.Fatalf("failed to make temp dir: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	cfgRESTMapper, err := meta.NewRESTMapperManager(tmpDir)
+	if err != nil {
+		t.Fatalf("failed to create REST mapper manager: %v", err)
+	}
+	proxyRESTMapper, err := meta.NewRESTMapperManager(tmpDir)
+	if err != nil {
+		t.Fatalf("failed to create proxy REST mapper manager: %v", err)
+	}
+
+	clientset := fake.NewSimpleClientset()
+	factory := informers.NewSharedInformerFactory(clientset, 0)
+	poolScopeResources := []schema.GroupVersionResource{endpointSliceGVR}
+
+	healthChecker := fakeHealthChecker.NewFakeChecker(map[*url.URL]bool{})
+	loadBalancer := remote.NewLoadBalancer("round-robin", []*url.URL{}, nil, nil, healthChecker, nil, context.Background().Done())
+	dsm := multiplexerstorage.NewDummyStorageManager(mockCacheMap())
+	cfg := &config.YurtHubConfiguration{
+		PoolScopeResources:       poolScopeResources,
+		RESTMapperManager:        cfgRESTMapper,
+		SharedFactory:            factory,
+		LoadBalancerForLeaderHub: loadBalancer,
+	}
+	rmm := multiplexer.NewRequestMultiplexerManager(cfg, dsm, healthChecker)
+
+	stopCh := make(chan struct{})
+	if ok := cache.WaitForCacheSync(stopCh, func() bool { return rmm.Ready(&endpointSliceGVR) }); !ok {
+		t.Fatal("multiplexer manager is not ready")
+	}
+
+	sp := NewMultiplexerProxy(rmm, proxyRESTMapper, make(<-chan struct{}))
+	w := httptest.NewRecorder()
+	req := newEndpointSliceListRequest("/apis/discovery.k8s.io/v1/endpointslices", nil)
+
+	assert.NotPanics(t, func() {
+		sp.ServeHTTP(w, req)
+	})
+}
+
 func TestShareProxy_ServeHTTP_LIST(t *testing.T) {
 	tmpDir, err := os.MkdirTemp("", "test")
 	if err != nil {

--- a/pkg/yurthub/proxy/multiplexer/multiplexerproxy_test.go
+++ b/pkg/yurthub/proxy/multiplexer/multiplexerproxy_test.go
@@ -162,11 +162,12 @@ func TestMultiplexerProxyServeHTTPReturnsOnGetReqScopeError(t *testing.T) {
 	rmm := multiplexer.NewRequestMultiplexerManager(cfg, dsm, healthChecker)
 
 	stopCh := make(chan struct{})
+	defer close(stopCh)
 	if ok := cache.WaitForCacheSync(stopCh, func() bool { return rmm.Ready(&endpointSliceGVR) }); !ok {
 		t.Fatal("multiplexer manager is not ready")
 	}
 
-	sp := NewMultiplexerProxy(rmm, proxyRESTMapper, make(<-chan struct{}))
+	sp := NewMultiplexerProxy(rmm, proxyRESTMapper, stopCh)
 	w := httptest.NewRecorder()
 	req := newEndpointSliceListRequest("/apis/discovery.k8s.io/v1/endpointslices", nil)
 

